File: Blueprints/BlueprintDefinitions/netcore3.1/SimpleDynamoDBFunction/template/test/BlueprintBaseName.1.Tests/FunctionTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

using Xunit;

using Amazon;
using Amazon.Lambda.Core;
using Amazon.Lambda.DynamoDBEvents;
using Amazon.Lambda.TestUtilities;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;

using BlueprintBaseName._1;

namespace BlueprintBaseName._1.Tests
{
    public class FunctionTest
    {
        [Fact]
        public void TestFunction()
        {
            DynamoDBEvent evnt = new DynamoDBEvent
            {
                Records = new List<DynamoDBEvent.DynamodbStreamRecord>
                {
                    new DynamoDBEvent.DynamodbStreamRecord
                    {
                        AwsRegion = "us-west-2",
                        Dynamodb = new StreamRecord
                        {
                            ApproximateCreationDateTime = DateTime.Now,
                            Keys = new Dictionary<string, AttributeValue> { {"id", new AttributeValue { S = "MyId" } } },
                            NewImage = new Dictionary<string, AttributeValue> { { "field1", new AttributeValue { S = "NewValue" } }, { "field2", new AttributeValue { S = "AnotherNewValue" } } },
                            OldImage = new Dictionary<string, AttributeValue> { { "field1", new AttributeValue { S = "OldValue" } }, { "field2", new AttributeValue { S = "AnotherOldValue" } } },
                            StreamViewType = StreamViewType.NEW_AND_OLD_IMAGES
                        }
                    }
                }
            };


            var context = new TestLambdaContext();
            var function = new Function();

            function.FunctionHandler(evnt, context);

            var testLogger = context.Logger as TestLambdaLogger;
			Assert.Contains("Stream processing complete", testLogger.Buffer.ToString());
        }  
    }
}

File: Blueprints/BlueprintDefinitions/vs2022/SimpleDynamoDBFunction-FSharp/template/test/BlueprintBaseName.1.Tests/FunctionTest.fs
namespace BlueprintBaseName._1.Tests

open Xunit
open Amazon.Lambda.TestUtilities

open Amazon.Lambda.DynamoDBEvents

open System
open System.Collections.Generic

open BlueprintBaseName._1


module FunctionTest =
    [<Fact>]
    let ``Test Reading DynamoDB Stream Event``() =
        let keys = dict [ ("id", DynamoDBEvent.AttributeValue(S = "MyId")) ]

        let newImage = dict [
                            ("field1", DynamoDBEvent.AttributeValue(S = "NewValue"))
                            ("field2", DynamoDBEvent.AttributeValue(S = "AnotherNewValue"))
                        ]

        let oldImages = dict [
                            ("field1", DynamoDBEvent.AttributeValue(S = "OldValue"))
                            ("field2", DynamoDBEvent.AttributeValue(S = "AnotherOldValue"))
                        ]

        let streamRecord =
            DynamoDBEvent.StreamRecord(
                ApproximateCreationDateTime = DateTime.Now,
                StreamViewType = "NEW_AND_OLD_IMAGES",
                Keys = Dictionary(keys),
                NewImage = Dictionary(newImage),
                OldImage = Dictionary(oldImages)
            )

        let ddbStreamRecords = [
            DynamoDBEvent.DynamodbStreamRecord(
                EventID = "id-foo",
                EventName = "INSERT",
                AwsRegion = "us-west-2",
                Dynamodb = streamRecord
            )
        ]

        let ddbEvent = DynamoDBEvent(Records = List(ddbStreamRecords))
        let lambdaFunction = Function()
        let testLogger = TestLambdaLogger()
        let context = TestLambdaContext(Logger = testLogger)
        lambdaFunction.FunctionHandler ddbEvent context

        Assert.Contains("Stream processing complete", testLogger.Buffer.ToString())

    [<EntryPoint>]
    let main _ = 0
